src/Parsers/ExpressionElementParsers.cpp

```diff
@@ -242,6 +243,39 @@ bool ParserIdentifier::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
 }
 
 
+bool ParserFormatIdentifier::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
```

Comment:
I don't understand why should the parser know about available formats. The factory needs to know how to handle it, not the parser.

Reply:
![format](https://github.com/ClickHouse/ClickHouse/assets/35292408/e9de61f8-ba2f-4bec-bb52-87efc57b6e43)
I want this effect, although it doesn't affect usability. I have removed it.